app/models/katello/host/subscription_facet.rb

```diff
@@ -34,6 +35,19 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_consumer_type(consumer_params)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.consumer_type = 'hypervisor'
+        elsif consumer_params.try(:[], 'facts').try(:[], 'virt.is_guest') == "true"
```

Comment:
I'm uncertain, but if subman sends bios facts, those might contain the vm uuid.
regarding searching, if its puppet facts, then its facts.virtual = true/false.